ts/screens/messages/paginated/MessagesHomeScreen.tsx

```diff
@@ -165,6 +167,34 @@ const MessagesHomeScreen = ({
     }
   });
 
+  useEffect(() => {
+    if (latestMessageOperation === undefined) {
+      return;
+    }
+
+    if (latestMessageOperation.isLeft()) {
+      switch (latestMessageOperation.value.operation) {
+        case "archive":
+          showToast(I18n.t("messages.operations.archive.failure"), "danger");
+          return;
+        case "restore":
+          showToast(I18n.t("messages.operations.restore.failure"), "danger");
+          return;
+      }
+    }
+
+    if (latestMessageOperation.isRight()) {
+      switch (latestMessageOperation.value) {
+        case "archive":
+          showToast(I18n.t("messages.operations.archive.success"), "success");
+          return;
+        case "restore":
+          showToast(I18n.t("messages.operations.restore.success"), "success");
+          return;
+      }
+    }
```

Comment:
```suggestion
    type toastData = {
      operation: MessageOperation;
      archive: string;
      restore: string;
      type: "success" | "danger";
    };
    pipe<typeof latestMessageOperation, toastData, void>(
      lmo =>
        lmo.fold<toastData>(
          l => ({
            operation: l.operation,
            archive: I18n.t("messages.operations.archive.failure"),
            restore: I18n.t("messages.operations.restore.failure"),
            type: "danger"
          }),
          r => ({
            operation: r,
            archive: I18n.t("messages.operations.archive.success"),
            restore: I18n.t("messages.operations.restore.success"),
            type: "success"
          })
        ),
      lmo => showToast(lmo[lmo.operation], lmo.type)
    )(latestMessageOperation);
```
this suggestion is about a more functional approach
Feel free to ignore since it is pretty the same

Some details:
`pipe` executes the functions in sequence, starting from the first to the last, by carrying the result to the next function
`function pipe<A, B, C>(ab: (a: A) => B, bc: (b: B) => C): (a: A) => C;`

Reply:
I liked it :) da4c2ee4